packages/terra-table/src/SelectableUtils.js

```diff
@@ -16,21 +29,77 @@ const initialSingleSelectRowIndex = (rows) => {
 };
 
 /**
-* Returns if the selected row index is already selected for SingleSelectableRows.
-*/
+ * Returns the indexes of the selected rows for MultiSelectableRows.
+ * To be used in the constructor, to set initial state.
+ */
+const initialMultiSelectRowIndexes = (rows, maxSelectionCount) => {
+  if (!rows || !rows.length) {
+    return [];
+  }
+  // Find the rows which are selected and are selectable
+  const selectedIndexes = [];
+  const validatedMaxSelectionCount = validatedMaxCountSelection(rows, maxSelectionCount);
+  for (let i = 0; i < rows.length; i += 1) {
+    if (rows[i].props.isSelected) {
+      selectedIndexes.push(i);
+
+      if (selectedIndexes.length >= validatedMaxSelectionCount) {
+        break;
+      }
+    }
+  }
+  return selectedIndexes;
+};
+
+/**
+ * Returns a new array, updated with the newIndex being added or removed from the existing.
+ */
+const updatedMultiSelectedIndexes = (currentIndexes, newIndex) => {
+  let newIndexes = [];
+  if (currentIndexes && currentIndexes.length) {
```

Comment:
Thinking we could rewrite this so it's not as nested. Something like:
```javascript
if(!Array.isArray(currentIndexes) || !currentIndexes.length ){
  newIndexes.push(newIndex);
  return newIndexes;
}
//same conditionals you had after the first if statement
```
Thoughts?

Reply:
If guard statements result in duplicate return statements I'd prefer documentation instead of additional lines, as the uglified code will still be smaller.

Reply:
To me, it's easier to follow the logic with a guard clause that is included at the top, even if there are multiple return statements. Guess it comes down to personal preference though, +1 either way.

Reply:
In this instance, I think its fine & is consistent with patterns seen in other components.

Reply:
I'm fine with making any changes, but I do want to bring up that I basically drew most of my coding and styling considerations from terra-list, which already has a MultiSelect component published. This function parallels [this function](https://github.com/cerner/terra-core/blob/master/packages/terra-list/src/SelectableUtils.js#L55) for terra-list. So I tried to not introduce my own stylistic tendencies here and opted to follow code that already made it through PR.

I'm fine with making the change, but since @yuderekyu already mentioned he's +1 either way, I'll leave it as is unless someone explicitly says to go ahead and change it.